aya/src/programs/mod.rs

```diff
@@ -439,8 +439,70 @@ impl<T: Link> ProgramData<T> {
             attach_prog_fd: None,
             btf_fd,
             verifier_log_level,
+            path: None,
         }
     }
+
+    pub(crate) fn from_bpf_prog_info(
+        name: Option<String>,
+        fd: RawFd,
+        path: &Path,
+        info: bpf_prog_info,
+    ) -> Result<ProgramData<T>, ProgramError> {
+        let attach_btf_id = if info.attach_btf_id > 0 {
+            Some(info.attach_btf_id)
+        } else {
+            None
+        };
+        let attach_btf_obj_fd = if info.attach_btf_obj_id > 0 {
+            let fd = bpf_btf_get_fd_by_id(info.attach_btf_obj_id).map_err(|io_error| {
+                ProgramError::SyscallError {
+                    call: "bpf_btf_get_fd_by_id".to_string(),
+                    io_error,
+                }
+            })?;
+            Some(fd as u32)
+        } else {
+            None
+        };
+
+        Ok(ProgramData {
+            name,
+            obj: None,
+            fd: Some(fd),
+            links: LinkMap::new(),
+            expected_attach_type: None,
+            attach_btf_obj_fd,
+            attach_btf_id,
+            attach_prog_fd: None,
+            btf_fd: None,
+            verifier_log_level: 0,
+            path: Some(path.to_path_buf()),
+        })
+    }
+
+    pub(crate) fn from_pinned_path<P: AsRef<Path>>(
+        path: P,
+    ) -> Result<ProgramData<T>, ProgramError> {
+        let path_string = std::ffi::CString::new(path.as_ref().to_str().unwrap()).unwrap();
```

Comment:
I think we should use to_string_lossy here

Reply:
also import types, don't use full qualified paths

Reply:
Ugh, so Path -> Cstring isn't exactly easy, but this is what I settled on.
```rust
CString::new(path.as_ref().as_os_str().to_string_lossy().as_bytes())
```